src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -909,10 +909,11 @@ public void setVelocity(@NotNull Vec velocity) {
     /**
      * Gets if the entity currently has a velocity applied.
      *
-     * @return true if velocity is not set to 0
+     * @return true if velocity is not set to 0 or the default velocity.
      */
     public boolean hasVelocity() {
-        return !velocity.isZero();
+        return !(velocity.isZero() ||
+                (velocity.x() == 0 && velocity.z() == 0 && Math.abs(velocity.y() + 1.568) < 0.001));
```

Comment:
Could you explain this (both here and in comment)? Look very magical to me.

Reply:
the entity does not have velocity if:
- the velocity is zero or
- the velocity is the default velocity that entities get from gravity (x=0=z, y=-1.568)

I invert that or statement to get the final result: Does the entity have velocity.
I'll add some comments to the statements as well

Reply:
I mean, how did you get this number?

Reply:
I printed out the velocity of the entity

Reply:
The number was also mentioned in the issue

Reply:
So it does not account for the gravity settings of the entity? it does not sound very robust

Reply:
Wait, entities can have gravity settings?

Reply:
drag is indeed not a global constant

Reply:
Ok, I'll try to find the formula for calculating the velocity from the gravity settings and use that.

Reply:
https://minecraft.fandom.com/wiki/Entity#Motion_of_entities

None of the numbers here match what the code produces. Is our gravity calculation different?

Reply:
The y velocity when onGround is equal to `-gravityAcceleration * tps * (1 - gravityDragPerTick)`